### ISSUE TITLE
RF03: Databricks  - STRUCT function alias is identified as column reference

### ISSUE THREAD

**Issue by PhilippLange** [sqlfluff/sqlfluff#6919]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have an issue with the behavior of linting the STRUCT function in databricks dialect.
I'm willing to attempt a fix as I contributed in the past.
Someone has to point me in the right direction though.

The issue occurs with the following statement:
```sql
-- input
SELECT
    tbl.a,
    STRUCT(
        SUM(tbl.b) as MEASUREMENT_VALUE,
        'MeasurementName' as MEASUREMENT_NAME
    ) AS measurement
FROM tbl
```

### Expected Behaviour

I expect the AS to be capitalized and the Coumn alias (within the struct) not to be altered:
```sql
-- expected output
SELECT
    tbl.a,
    STRUCT(
        SUM(tbl.b) AS MEASUREMENT_VALUE,
        'MeasurementName' AS MEASUREMENT_NAME
    ) AS measurement
FROM tbl
```

### Observed Behaviour

Instead the column alias is identified as a reference and made explicit and lowerecased while the AS is not capitalized:
```sql
-- actual output
SELECT
    tbl.a,
    STRUCT(
        SUM(tbl.b) as tbl.measurement_value,
        'MeasurementName' as tbl.measurement_name
    ) AS measurement
FROM tbl
```

```sh
L:   4 | P:  23 | RF03 | Unqualified reference 'MEASUREMENT_VALUE' found in
                       | single table select. [references.consistent]
L:   5 | P:  30 | RF03 | Unqualified reference 'MEASUREMENT_NAME' found in single
                       | table select. [references.consistent]
```

### How to reproduce

Lint the following statement:
```sql
SELECT
    tbl.a,
    STRUCT(
        SUM(tbl.b) as MEASUREMENT_VALUE,
        'MeasurementName' as MEASUREMENT_NAME
    ) AS measurement
FROM tbl
```

### Dialect

databricks

### Version

3.4.0

### Configuration

[sqlfluff]
dialect = databricks

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
